### DISCUSSION TITLE
ReCaptcha V3 refusé mais création de compte possible

### DISCUSSION THREAD

**Discussion by ththieul** [AmauriC/tarteaucitron.js#908] [Q&A] [answered]
Bonjour !

Je rencontre un soucis avec TaC qui est que mes actions sur le site fonctionne quand même alors que le cookie de reCAPTCHA V3 est refusé par le biais de TaC. Je voulais donc savoir ce qui clochait ou s'il était possible de m'aiguiller sur mon problème svp ?

voici le script permettant de mettre en place le reCAPTCHA sur la création de compte sur le site : 

```
<script>
    
        $(\'#compte\').submit(function(event) {
            event.preventDefault();
          
            grecaptcha.ready(function() {
                grecaptcha.execute(\''.CAPTCHA_SITE_KEY.'\', {action: \'account\'}).then(function(token) {
                    var recaptcha = document.getElementById(\'tokencptchacc\');
                    recaptcha.value = token;
                    $(\'#compte\').unbind(\'submit\').submit();
                });
            });
        });
</script>
```

Voici le code en backend pour vérifier la validité du token CAPTCHA : 

```
$data = array('secret' => CAPTCHA_SECRET_KEY, 'response' => filter_input(INPUT_POST, $input_captcha, FILTER_SANITIZE_STRING));
		$options = array(
			'http' => array(
				'header'  => "Content-type: application/x-www-form-urlencoded\r\n",
				'method'  => 'POST',
				'content' => http_build_query($data)
		  )
		);
		$context  = stream_context_create($options);
		$response = file_get_contents(CAPTCHA_VERIFY_URL, false, $context);
		$responseKeys = json_decode($response,true);
```

Voici le code pour l'ajout de reCAPTCHA à TaC :

```
<script type="text/javascript">
                        tarteaucitron.user.recaptchaapi = \''.CAPTCHA_SITE_KEY.'\';
                        (tarteaucitron.job = tarteaucitron.job || []).push("recaptcha");
</script>
```

J'ai l'impression que l'exemple fourni pour l'intégration du reCAPTCHA par TaC est sur une version 1 ou 2 du reCAPTCHA mais ne correspond pas du tout à la V3.

Merci pour votre aide !

**Comment by AmauriC** [comment, marked as answer]
Sur le site il n'y a plus le marqueur reCaptcha ?

En vérifiant si `grecaptcha` existe ou en utilisant `tarteaucitron.state.recaptcha` tu peux conditionner l'envoi du formulaire et afficher un message d'erreur.

**Reply by ththieul** [reply]
Qu'appelez-vous marqueur reCaptcha ? 

Est ce que c'est bien ce que représente l'image ci dessous ?
![image](https://user-images.githubusercontent.com/57672787/158818177-4e66c007-dcd4-4bb9-bad6-dc30dba4fabe.png)
Sinon champ qui récupère le token Captcha ?

Oui, avec `tarteaucitron.state.recaptcha` je peux conditionner l'envoie du formulaire et arrêter moi-même le service reCaptcha à l'appel.
J'imagine que c'est ce que fais TaC quand il y a le marqueur.

**Reply by AmauriC** [reply]
Je parlais du marqueur HTML fourni par Google.

Sur la capture recaptcha est activé sur tarteaucitron et le bandeau recaptcha disparait quand il est désactivé?
Si oui le problème est que le formulaire est validé avec ou sans recaptcha valide : /

**Reply by ththieul** [reply]
C'était belle est bien le marqueur reCaptcha fourni par Google qui restait en tête de code. Maintenant, cela s'active et se désactive bien avec TaC.

Merci beaucoup pour votre aide ! =)
